config/validator.go

```diff
@@ -142,3 +142,40 @@ func RequestHeaderContentTypeValidator(req http.Request, requiredContentType str
 	}
 	return 0
 }
+
+
+// DomainsEntryLogicalValidator checks the correctness of "Domains" part of configuration file
+func (c *YamlConfig) DomainsEntryLogicalValidator() (valid bool, validationErrors map[string][]error) {
+	errList := make([]error, 0)
+	var domains []string
+	for _, regionConf := range c.Regions {
+		for _, domain := range regionConf.Domains {
+			for _, alreadyDefinedDomain := range domains {
+				if domain == alreadyDefinedDomain {
+					continue
+				}
+				if strings.HasSuffix(domain, "." + alreadyDefinedDomain) {
+					errList = append(
+						errList,
+						fmt.Errorf("Invalid domain %s! There is a domain %s already defined", domain, alreadyDefinedDomain))
+					continue
+				} else if strings.HasSuffix(alreadyDefinedDomain, "." + domain) {
```

Comment:
this if and one above could be merged into single case, with same error "domain %q conflicts with %q"